walk-in-interview/src/main/java/com/google/utils/ServletUtils.java

```diff
@@ -0,0 +1,142 @@
+     */
+    public static String getStringParameter(HttpServletRequest request, String name, String defaultValue) {
+        String value = request.getParameter(name).trim();
+        if (value == null || value.isEmpty()) {
+            return defaultValue;
+        }
+        return value;
+    }
+
+    /**
+     * Gets the integer parameter from html form input.
+     *
+     * @param request Http request.
+     * @param parameterName Input id in html element.
+     * @param defaultValue Default integer.
+     * @return Parsed integer or default value if exception occur
+     */
+    public static int getIntParameter(HttpServletRequest request, String parameterName, int defaultValue) {
+        String resultStr = request.getParameter(parameterName).trim();
+
+        try {
+            int result = Integer.parseInt(resultStr);
+
+            return result;
+        } catch (NumberFormatException e) {
+            // TODO(issue/12): error handling; can consider to add logging to log the error
+
+            return defaultValue;
+        }
+    }
+
+    /**
+     * Gets the long parameter from html form input.
+     *
+     * @param request Http request.
+     * @param parameterName Input id in html element.
+     * @param defaultValue Default integer.
+     * @return Parsed integer or default value if exception occur
+     */
+    public static long getLongParameter(HttpServletRequest request, String parameterName, long defaultValue) {
+        String resultStr = request.getParameter(parameterName).trim();
+
+        try {
+            long result = Integer.parseInt(resultStr);
+
+            return result;
+        } catch (NumberFormatException e) {
+            // TODO(issue/12): error handling; can consider to add logging to log the error
+
+            return defaultValue;
+        }
+    }
+
+    /**
+     * Gets the float parameter from html form input.
+     *
+     * @param request Http request.
+     * @param parameterName Input id in html element.
+     * @param defaultValue Default integer.
+     * @return Parsed float or default value if exception occur
+     */
+    public static float getFloatParameter(HttpServletRequest request, String parameterName, float defaultValue) {
+        String resultStr = request.getParameter(parameterName).trim();
+
+        try {
+            float result = Float.parseFloat(resultStr);
+
+            return result;
+        } catch (NumberFormatException e) {
+            // TODO(issue/12): error handling; can consider to add logging to log the error
+
+            return defaultValue;
+        }
+    }
+
+    /**
+     * Gets the LocalDate parameter from html form input.
+     * @param request Http request.
+     * @param parameterName Input id in html element.
+     * @param defaultValue Default integer.
+     * @return Parsed date or default value if exception occur
+     */
+    public static Date getDateParameter(HttpServletRequest request,
```

Comment:
Should we have test for these methods? Especially when the param is wrongly formatted.

Reply:
So far this method is not called yet, so maybe test it when it is called?